tests/test_plan.py

```diff
@@ -459,3 +460,29 @@ def test_plan_overrides(recursive_overrides):
         assert plan2 == plan
         assert (plan == [(A, {}), (B, {}), (C, {'a': A, 'b': B}), (D, {'a': A, 'c': C})] or
                 plan == [(B, {}), (A, {}), (C, {'a': A, 'b': B}), (D, {'a': A, 'c': C})])
+
+
+@pytest.mark.skipif(sys.version_info < (3, 9), reason="No Annotated support in Python < 3.9")
+def test_plan_annotations():
+    class MyFunc:
+        def __call__(self, b: Annotated[B, 42]):
+            pass
+
+    func = MyFunc()
+    plan = andi.plan(func, is_injectable={B})
+    assert plan == [(Annotated[B, 42], {}), (func, {'b': Annotated[B, 42]})]
+
+
+@pytest.mark.skipif(sys.version_info < (3, 9), reason="No Annotated support in Python < 3.9")
+def test_plan_annotations_duplicate():
+    class MyFunc:
+        def __call__(self, b: Annotated[B, 42], b2: Annotated[B, 43]):
+        # def __call__(self, b: B, b2: B):
+            pass
+
+    func = MyFunc()
+    plan = andi.plan(func, is_injectable={B})
+    assert plan == [
+        (Annotated[B, 42], {}),
+        (func, {'b': Annotated[B, 42], 'b2': Annotated[B, 42]}),
```

Comment:
This fails as it returns `'b2': Annotated[B, 43]`, taken straight out of the function signature. It should be possible to make the code do what the text expects, but I don't know what should it ideally expect.